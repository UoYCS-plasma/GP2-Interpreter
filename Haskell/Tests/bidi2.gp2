Main = r1

r1 (x,y:int) [
	(n1, x)
	(n2, y)
	| (e1 (B), n1, n2, empty)
] => [
	(n1, x)
	(n2, y)
	| (e1 (B), n2, n1, empty)
] interface = {n1, n2}
