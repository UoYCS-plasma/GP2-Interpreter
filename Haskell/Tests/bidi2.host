[
	(n1, 1)
	(n2, 2)
|
	(e1, n1, n2, empty)
]
