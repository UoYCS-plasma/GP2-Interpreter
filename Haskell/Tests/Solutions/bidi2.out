[
 (n1, 2)
 (n2, 1)
|
 (e1, n2, n1, empty)
]
